twitch4j/src/main/java/com/github/twitch4j/TwitchClientHelper.java

```diff
@@ -578,103 +569,47 @@ private void updateListener(BooleanSupplier stopCondition, AtomicReference<Futur
     }
 
     /**
-     * Updates {@link ExponentialBackoffStrategy#getBaseMillis()} for each of the independent listeners (i.e. stream status and followers)
-     *
-     * @param threadRate the maximum <i>rate</i> of api calls per second
-     */
-    public void setThreadRate(long threadRate) {
-        this.setThreadDelay(1000 / threadRate);
-    }
-
-    /**
-     * Updates {@link ExponentialBackoffStrategy#getBaseMillis()} for each of the independent listeners (i.e. stream status and followers)
-     *
-     * @param threadDelay the minimum milliseconds <i>delay</i> between each api call
+     * Initiates the clip creation listener execution
      */
-    public void setThreadDelay(long threadDelay) {
-        final UnaryOperator<ExponentialBackoffStrategy> updateBackoff = old -> {
-            ExponentialBackoffStrategy next = old.toBuilder().baseMillis(threadDelay).build();
-            next.setFailures(old.getFailures());
-            return next;
-        };
-
-        this.liveBackoff.getAndUpdate(updateBackoff);
-        this.followBackoff.getAndUpdate(updateBackoff);
-    }
-
-    /**
-     * Get cached information for a channel's stream status and follower count.
-     * <p>
-     * For this information to be valid, the respective event listeners need to be enabled for the channel.
-     * <p>
-     * For thread safety, the setters on this object should not be used; only getters.
-     *
-     * @param channelId The ID of the channel whose cache is to be retrieved.
-     * @return ChannelCache in an optional wrapper.
-     */
-    public Optional<ChannelCache> getCachedInformation(String channelId) {
-        return Optional.ofNullable(channelInformation.getIfPresent(channelId));
+    private void runRecursiveClipCheck() {
+        runRecursiveCheck(clipEventFuture, executor, new ArrayList<>(listenForClips), clipBackoff, this::runRecursiveClipCheck, clipEventTask);
     }
 
-    /**
-     * Close
-     */
-    public void close() {
-        final Future<?> streamStatusFuture = this.streamStatusEventFuture.getAndSet(null);
-        if (streamStatusFuture != null)
-            streamStatusFuture.cancel(false);
-
-        final Future<?> followerFuture = this.followerEventFuture.getAndSet(null);
-        if (followerFuture != null)
-            followerFuture.cancel(false);
-
-        listenForGoLive.clear();
-        listenForFollow.clear();
-        channelInformation.invalidateAll();
+    private List<Clip> getClips(String channelId, Instant startedAt, Instant endedAt) {
+        return PaginationUtil.getPaginated(
+            cursor -> {
+                final HystrixCommand<ClipList> commandGetClips = twitchHelix.getClips(null, channelId, null, null, cursor, null, MAX_LIMIT, startedAt, endedAt);
+                try {
+                    ClipList result = commandGetClips.execute();
+                    clipBackoff.get().reset(); // successful api call
+                    return result;
```

Comment:
currently no delay is being applied within paginated clip calls for a given channel (but it's very rare to hit pagination even on the most popular broadcasts)